examples/06-Multiphysics/Maxwell3D_Icepak_2Way_Coupling.py

```diff
@@ -0,0 +1,239 @@
+solution = report.get_solution_data()
+resistance = solution.data_magnitude()[0]
+
+m3d.logger.info("*******Coil resistance BEFORE temperature feedback =  {:.2f}mOhm".format(1000*resistance))
+
+###############################################################################
+# Icepak design
+# ~~~~~~~~~~~~~
+# Insert icepak design and copy solid objects from Maxwell, modify region dimensions
+
+ipk = pyaedt.Icepak(designname=icepak_design_name)
+ipk.copy_solid_bodies_from(m3d, no_pec=False)
+
+# Set domain dimensions suitable for natural convection using the diameter of the coil
+ipk.modeler["Region"].delete()
+l=coil.bounding_dimension[0]
+ipk.modeler.create_region(0, False)
+ipk.modeler.edit_region_dimensions([l/2, l/2, l/2, l/2, l*2, l])
+
+###############################################################################
+# Map coil losses
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Map ohmic losses in coil
+
+ipk.assign_em_losses(
+    designname="1 Maxwell",
+    setupname=m3d.setups[0].name,
+    sweepname="LastAdaptive",
+    object_list=["Coil"],
+)
+
+###############################################################################
+# Boundary conditions
+# ~~~~~~~~~~~~~~~~~~~
+# Assign opening
+
+faces = ipk.modeler["Region"].faces
+face_names = [face.id for face in faces]
+ipk.assign_free_opening(face_names, boundary_name="Opening1")
+
+###############################################################################
+# Icepak solution setup
+# ~~~~~~~~~~~~~~~~~~~
+
+solution_setup = ipk.create_setup()
+solution_setup.props["Convergence Criteria - Max Iterations"] = 50
+solution_setup.props["Flow Regime"] = "Turbulent"
+solution_setup.props["Turbulent Model Eqn"] = "ZeroEquation"
+solution_setup.props["Radiation Model"] = "Discrete Ordinates Model"
+solution_setup.props["Include Flow"] = True
+solution_setup.props["Include Gravity"] = True
+solution_setup.props["Solution Initialization - Z Velocity"] = "0.0005m_per_sec"
+solution_setup.props["Convergence Criteria - Flow"] = 0.0005
+solution_setup.props["Flow Iteration Per Radiation Iteration"] = "5"
+
+
+###############################################################################
+# Add 2-way coupling and solve the project
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+ipk.assign_2way_coupling()
+ipk.analyze_setup(name=solution_setup.name)
+
+###############################################################################
+# Post-processing
+# ~~~~~~~~~~~~~~~~~~~
+# Plot temperature on the object surfaces
+
+surface_list = []
+for name in ["Coil", "Core"]:
+    surface_list.extend(ipk.modeler.get_object_faces(name))
+
+surf_temperature = ipk.post.create_fieldplot_surface(
+    surface_list,
+    quantityName="SurfTemperature",
+    plot_name="Surface Temperature"
+)
+
+velocity_cutplane = ipk.post.create_fieldplot_cutplane(
+        objlist="Global:XZ",
+        quantityName="Velocity Vectors",
+        plot_name="Velocity Vectors"
+    )
+
+surf_temperature.export_image()
```

Comment:
just as a comment you could also leverage PyVista with the method 
`plot_field_from_fieldplot`
where you have more options.